R/kernel.r

```diff
@@ -242,6 +243,92 @@ complete = function(request) {
         cursor_end = start_position + nchar(c.info$token)))
 },
 
+inspect = function(request) {
+    make_section_header_plain <- function (txt) paste0("# ", txt, ":\n")
+    make_section_header_html <- function (txt) paste0("<h1>", txt, ":</h1>\n")
+    # 5.0 protocol:
+    code <- request$content$code
+    cursor_pos <- request$content$cursor_pos
+
+    # Get token under the cursor_pos.
+    # There may be better ways to do that than this.
+    token <- ""
+    for (i in seq(cursor_pos, nchar(code))) {
```

Comment:
The reason for the loop is that `tools:::.guessTokenFromLine()` checks only characters before `cursor_pos`. It is necessary to check characters after `cursor_pos` in some way if we want to get the correct token on invoking `inspect_request` when the cursor is within a token.
Some combination of string manipulate functions may do the trick, but I thought the loop is easier to understand.

But I must admit that my implementation causes a problem when the cursor is above punctuations, white spaces or so. I should finish the loop if the first call of function returns an empty string. Because `tools:::.guessTokenFromLine()` returns an empty string if `cursor_pos` is on the characters not used in tokens.

Or, is there a tokenizer function which fits this purpose more? If not, I think of putting the code to finish the loop on the condition I mentioned above. How do you think?